src/fastpam1.cpp

```diff
@@ -0,0 +1,193 @@
+/**
+ * @file fastpam1.cpp
+ * @date 2021-08-03
+ *
+ * This file contains the primary C++ implementation of the FastPAM1 code.
+ *
+ */
+#include "kmedoids_algorithm.hpp"
+
+#include <carma.h>
+#include <armadillo>
+#include <unordered_map>
+#include <regex>
+
+/**
+ * \brief Runs naive FastPAM1 algorithm.
+ *
+ * Run the naive FastPAM1 algorithm to identify a dataset's medoids.
+ *
+ * @param input_data Input data to find the medoids of
+ */
+void KMedoids::fit_fastpam1(arma::mat input_data) {
+  data = input_data;
+  data = arma::trans(data);
+  arma::rowvec medoid_indices(n_medoids);
+  // runs build step
+  KMedoids::build_fastpam1(data, medoid_indices);
+  steps = 0;
```

Comment:
appears unused; rm?

Reply:
Ah; that function uses a lot of globals. Instead, we should make those member variables. It's ok for this PR and I've updated in #110 for us to fix later.